src/main.lua

```diff
@@ -130,13 +130,14 @@ end, function(msg)
 
 	local msgTimestamp = tonumber(msg.Timestamp)
 	print("Pruning state at timestamp: " .. msgTimestamp)
-	-- TODO: we should copy state here and restore if tick fails, but that requires larger memory - DO NOT DO THIS UNTIL WE START PRUNING STATE of epochs and distributions
-	local status, resultOrError = pcall(tick.pruneState, msgTimestamp, msgId)
 	local previousState = {
 		Vaults = utils.deepCopy(Vaults),
 		GatewayRegistry = utils.deepCopy(GatewayRegistry),
 		NameRegistry = utils.deepCopy(NameRegistry),
+		Epochs = utils.deepCopy(Epochs),
+		-- TODO: add vaults and balances
 	}
+	local status, resultOrError = pcall(tick.pruneState, msgTimestamp, msgId)
```

Comment:
I'm glad you fixed this because I was confused why we'd stash the previous state AFTER we already made the change 😅